### PR TITLE
bots: Add fedora-29 image

### PR DESCRIPTION
**Pull request by martinpitt**
Copy fedora-28 naughty overrides which are still relevant.

Enable this image for every PR test, as we need to follow changes there
and spot issues like #9889 early.

 * [x] image-refresh fedora-29
 - [x]  work around docker networking regression: https://bugzilla.redhat.com/show_bug.cgi?id=1622543
 - [x] fix cockpit-ssh crash: https://bugs.libssh.org/T96 (temporarily worked around with patch from #9904 - remove before landing)
 - [x] fix `test-httpstream 13 /http-stream/tls/authority-bad` unit test failure; worked around for now, fixed in latest glib-networking

### CODE REVIEW

**Comment by martinpitt** [comment]
Local build fails on:
```
+ docker build -t cockpit/base /var/tmp/cockpit-base
Sending build context to Docker daemon 4.608 kB
Step 1/3 : FROM fedora:27
Trying to pull repository docker.io/library/fedora ... 
sha256:759eb1437bdcdf1a5aac3ba559b28ea6acbfbc4da6b49ff856ac53c3a6789c26: Pulling from docker.io/library/fedora
c8ae8b35783e: Pulling fs layer
c8ae8b35783e: Verifying Checksum
c8ae8b35783e: Download complete
c8ae8b35783e: Pull complete
Digest: sha256:759eb1437bdcdf1a5aac3ba559b28ea6acbfbc4da6b49ff856ac53c3a6789c26
Status: Downloaded newer image for docker.io/fedora:27
 ---> 1bf38ba7ab50
Step 2/3 : ADD setup.sh /setup.sh
 ---> bebf047e0c63
Removing intermediate container 42b48c095e2c
Step 3/3 : RUN /setup.sh
 ---> Running in 79224f62c086
DNF version: 2.7.5
cachedir: /var/cache/dnf
Cannot download 'https://mirrors.fedoraproject.org/metalink?repo=updates-released-f27&arch=x86_64': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=updates-released-f27&arch=x86_64 [Could not resolve host: mirrors.fedoraproject.org].
Error: Failed to synchronize cache for repo 'updates'
```
This isn't just transient, it repeats in my local VM. curl right in there works, but not inside the fedora:27 container.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-4np2l.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9916-20180827-124242/

**Comment by cockpituous** [comment]
image-refresh fedora-29 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-29-20180827-140101

**Comment by martinpitt** [comment]
@Gundersanne: Hah, and here we are being blocked on that cockpit-ssh crash again. Was there any other workaround than your closed PR #9904?

**Comment by Gundersanne** [comment]
@martinpitt So once you answer 1 answer, you have to answer them all, so I'm afraid not :/ They said libssh-0.8.2 was going to be released today though.

**Comment by martinpitt** [comment]
To unblock testing and discovering/investigating  other regressions, I temporarily stole the patch from #9904. This doesn't need to/shouldn't actually land, though.

**Comment by martinpitt** [comment]
@Gundersanne: So this proves that the fix from #9904 at fixes the crash :) (See above, it's temporary). However, there's another unit test that fails:

```
DEBUG: MESSAGE: cockpit-bridge: requires: package has an unknown requirement: unknown
DEBUG: cockpit-bridge-Message: 06:01:54.628: https://localhost:41537/test: couldn't read: Server required TLS certificate
DEBUG: **
DEBUG: cockpit-bridge:ERROR:src/bridge/test-httpstream.c:816:test_tls_authority_bad: Got unexpected message: https://localhost:41537/test: couldn't read: Server required TLS certificate instead of cockpit-bridge-Message: *Unacceptable TLS certificate
[...]
FAIL: test-httpstream 13 /http-stream/tls/authority-bad
```

Could you please have a look? This also seems to be related to a newer GnuTLS.

**Comment by Gundersanne** [comment]
Using glib-networking 2.57.92 works just fine, so temporarily loosening the test's asserts/text matching should be fine.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-6qcf6.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9916-20180903-154242/

**Comment by cockpituous** [comment]
verify/fedora-29
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testContainerProblems (check_docker.TestDocker)
#
[0903/155652.342850:ERROR:gpu_process_transport_factory.cc(1017)] Lost UI shared context.

DevTools listening on ws://127.0.0.1:9845/devtools/browser/a8f95693-14fb-420f-80a8-147aeb064425
[0903/155652.389272:ERROR:zygote_host_impl_linux.cc(267)] Failed to adjust OOM score of renderer with pid 26800: Permission denied (13)
[0903/155652.504994:ERROR:zygote_host_impl_linux.cc(267)] Failed to adjust OOM score of renderer with pid 26850: Permission denied (13)
> warning: Failed to query RHEL subscription status: 
Unexpected journal messages:
audit: type=1400 audit(1535990116.290:91): avc:  denied  { create } for  pid=592 comm="ebtables" scontext=system_u:system_r:firewalld_t:s0 tcontext=system_u:system_r:firewalld_t:s0 tclass=rawip_socket permissive=0
Wrote JS log to TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.js.log
Journal extracted to TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.log
Core dumps downloaded to /build/cockpit/TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.core
Wrote screenshot to TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.png
Wrote HTML dump to TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.html
Wrote JS log to TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.js.log
Journal extracted to TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.log
Core dumps downloaded to /build/cockpit/TestDocker-testContainerProblems-fedora-29-127.0.0.2-2401-FAIL.core
Traceback (most recent call last):
  File "/build/cockpit/test/common/testlib.py", line 686, in tearDown
    self.check_journal_messages()
  File "/build/cockpit/test/common/testlib.py", line 859, in check_journal_messages
    raise Error(first)
testlib.Error: audit: type=1400 audit(1535990116.290:91): avc:  denied  { create } for  pid=592 comm="ebtables" scontext=system_u:system_r:firewalld_t:s0 tcontext=system_u:system_r:firewalld_t:s0 tclass=rawip_socket permissive=0

not ok 185 testContainerProblems (check_docker.TestDocker) # duration: 128s
```
First occurrence: 2018-09-03T15:57:04.479520 | revision cfb78e048479318d1e96353d6d98c7858fc73ddb, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-9916-20180903-152727-cfb78e04-verify-fedora-29/log.html)
Times recorded: 1
Latest occurrences:

- 2018-09-03T15:57:04.479520 | revision cfb78e048479318d1e96353d6d98c7858fc73ddb, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-9916-20180903-152727-cfb78e04-verify-fedora-29/log.html)

**Comment by martinpitt** [comment]
Current F-29 has libssh 0.8.2 and glib-networking 2.58.0, which should fix these issues. So the two hacks commits can be dropped, and I ticked the two TODO items.

**Comment by cockpituous** [comment]
image-refresh fedora-29 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-29-20180903-162727

**Comment by martinpitt** [comment]
The other "HACK" should  be dropped as well, and the "new" and "updated" image be squashed into one.

**Comment by Gundersanne** [comment]
Neither of them have been pushed downstream yet, they're still in testing.

**Comment by Gundersanne** [comment]
The remaining failures all seem to be flakes, but very persistent flakes :/ The ovirt ones all succeeded this round, but the kubernetes ones failed again.

**Comment by martinpitt** [comment]
Same in #10001 - the ovirt and kubevirt tests are currently almost impossible to get green both together. We have to fix them this week, one way or the other.

File: test/verify/check-storage-resize [hunk 1]
```diff
@@ -80,6 +80,7 @@ class TestStorage(StorageCase):
             size = int(m.execute("df -k --output=size %s | tail -1" % mountpoint).strip())
             assert (size > 300000)
         else:
+            sit()
```
Comment (martinpitt) [inline test/verify/check-storage-resize]: Debugging leftover

File: test/verify/check-packagekit [hunk 2]
```diff
@@ -670,7 +670,7 @@ class TestUpdates(PackageCase):
 
 
 @skipImage("Image uses OSTree", "continuous-atomic", "fedora-atomic", "rhel-atomic")
-@skipImage("No subscriptions", "centos-7", "debian-stable", "debian-testing", "fedora-28", "fedora-i386", "fedora-testing", "ubuntu-1604", "ubuntu-stable")
+@skipImage("No subscriptions", "centos-7", "debian-stable", "debian-testing", "fedora-27", "fedora-28", "fedora-29", "fedora-i386", "fedora-testing", "ubuntu-1604", "ubuntu-stable")
```
Comment (martinpitt) [inline test/verify/check-packagekit]: fedora-27 is no more, this should be dropped again.

**Pull request merged**